metamath.c

```diff
@@ -1138,7 +1138,7 @@ void command(int argc, char *argv[])
           }
         }
         /* Do the operating system command */
-        (void)system(str1);
+        (void)!system(str1);

```

Comment:
The result of system() is not used.  To suppress warnings, and to indicate this is deliberate, a cast to void is performed. https://stackoverflow.com/questions/689677/why-cast-unused-return-values-to-void

If there is a reason to invert the result this should be noted in a comment.

Reply:
Will do. The reason is because of https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66425 : gcc does not respect the cast-to-void trick, and it is not clear whether this is intentional on their part but it doesn't look like it's going to be fixed any time soon.

An alternative trick is `if(system(str1)) {}`, not sure if that's any better.

Reply:
Fine with me.

Reply:
If we want a coding style, perhaps we could get ambitious and shoot for "don't ignore the return value from system" (because it could fail if out of memory, out of file descriptors, etc)?

I mean, we have gotten by with this code for a long time, so I'm not saying this is necessarily our top priority, just that.... well putting in a simple error check might not kill us.

Reply:
@jkingdon In this case, the call to `system` is because the user explicitly attempted to call an external program. The program is basically acting like bash here, so failures are expected and should not crash the program, although arguably they should be displayed (even though bash itself doesn't, it just stashes the result in the `$?` variable).

Reply:
Unless we have something analogous to $? I'd say display a message, yes.